reports/partB_dataAnalysis/dataAnalysisReport.Rmd

```diff
@@ -321,31 +379,119 @@ kableExtra::kable(t, caption = "Categoriesed circuits - XX check carefully! XX")
+sampleDT <- sampleDT[, eecaCircuit := ifelse(circuitLabel == "Heat Pump" |
+                                                       circuitLabel == "Theatre Heat Pump" |
+                                                       circuitLabel == "Upstairs Heat Pumps" |
+                                                       circuitLabel == "Heating",
+                                                    "Heat Pump or Heating", 
+                                                    "XX_Other") # default
+                             ]
+
+# these match if the string is in the circuit label (i.e. circuit label could contain other words)
+# Be careful with these
+sampleDT <- sampleDT[, eecaCircuit := ifelse( circuitLabel == "Oven" |
+                                                        circuitLabel == "Range" |
+                                                        circuitLabel == "Wall Oven" |
+                                                        circuitLabel %like% "Oven &" |
+                                                        circuitLabel %like% "Oven,", # if the circuit = Oven only
+                                                    "Oven", eecaCircuit)]
+sampleDT <- sampleDT[, eecaCircuit := ifelse(circuitLabel %like% "Hot water", 
+                                                    "Hot water", eecaCircuit)]
+sampleDT <- sampleDT[, eecaCircuit := ifelse(circuitLabel %like% "Hot Water", 
+                                                    "Hot water", eecaCircuit)]
+sampleDT <- sampleDT[, eecaCircuit := ifelse(circuitLabel %like% "Lighting" |
+                                                       circuitLabel %like% "Lights", 
+                                                    "Lighting", eecaCircuit)]
+sampleDT <- sampleDT[, eecaCircuit := ifelse(circuitLabel %like% "imputedTotalDemand_circuitsToSum_v1.1", 
+                                                    "Total", eecaCircuit)]
+
+sampleDT <- sampleDT[, summeanPowerW := NA]
+
+sampleShortDT <- sampleDT[, .(summeanPowerW = sum(meanPowerW, na.rm = TRUE), 
+                              nobs=.N), keyby=.(eecaCircuit, date, season, year, obsHalfHour)]#This is the aggregated Wattage for each circuit month etc. see keyby in order to conduct sample coincident peak analysis 
+
+
+sampleShortDT <- sampleShortDT[, averagePowerW := summeanPowerW/nobs]#We have to consider that some of the obs have only one obs, some have up to 200. We need to take the average to identify the top 20 entries (their time of occurance)
 
+
+sample2015DT <- sampleShortDT[ !(year %in% c(2014, 2016, 2017, 2018))]#Getting rid of years that are not necessary for this analysis
+
+sort(sample2015DT$averagePowerW, decreasing = TRUE)#Now identufy these top20 times in the original data table and conduct analysis again
+
+####Filtering the top20 peak half-hours based on our sample. XXThis is hard-coded - is there a better way to do it??XX
```

Comment:
there has to be!

Reply:
probably something like dateTime_halfHour %IN% x where x is the list of dateTime_halfHours